typer/cli.py

```diff
@@ -272,10 +273,13 @@ def get_docs_for_click(
     return docs
 
 
-def _parse_html(input_text: str) -> str:
-    if not has_rich:  # pragma: no cover
-        return input_text
-    return rich_utils.rich_to_html(input_text)
+def _parse_html(ctx: click.Context, input_text: str) -> str:
+    rich_markup_mode = None
+    if hasattr(ctx, "obj") and isinstance(ctx.obj, dict):
+        rich_markup_mode = ctx.obj.get("TYPER_RICH_MARKUP_MODE", None)
+    if has_rich and rich_markup_mode and rich_markup_mode == "rich":  # pragma: no cover
+        return rich_utils.rich_to_html(input_text)
+    return input_text
```

Comment:
🤔 I think we should move the computation of `rich_markup_mode` in `get_docs_for_click` so we don't do it more than necessary (and we keep this function simple)

Reply:
Yea fair enough, I had it there first 😁  Somehow that felt like cluttering the `get_docs_for_click` function, and conceptually it's nicer to pass around `ctx` everywhere. But I agree that it means repeating the same computation (though they should be simple, fast checks/retrievals)